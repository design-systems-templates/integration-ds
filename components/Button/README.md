# @integration-ds/button

Welcome to my `Button` component.

## Installation

```sh
npm i @integration-ds/button
# or with yarn
yarn add @integration-ds/button
```

Then to use the component in your code just import it!

```js
import Button from '@integration-ds/button';
// and with css-modules
import '@integration-ds/button/dist/main.css';
```
