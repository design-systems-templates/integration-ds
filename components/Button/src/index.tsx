import React from 'react';
import styles from './Button.css';

interface ButtonProps {
  /** The value for the component to display */
  value?: string;
}

/** A simple component. */
export const Button = (props: ButtonProps) => {
  return <button type="button" className={styles.button}>This is a Button component {props.value}</button>;
}

export default Button;
