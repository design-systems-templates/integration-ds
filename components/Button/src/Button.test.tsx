import React from 'react';
import { render } from '@testing-library/react';
import Button from '.';

describe('Button', () => {
  test('It matches the snapshot', () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });  
});