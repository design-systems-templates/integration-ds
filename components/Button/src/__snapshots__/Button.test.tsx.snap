// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button It matches the snapshot 1`] = `
<div>
  <div
    class="button"
  >
    This is a Button component 
  </div>
</div>
`;
